### PR TITLE
Allow KafkaClient to configure source_address

### PR DESCRIPTION
**Pull request by emmett9001**
This pull request adds a kwarg `source_address` to `KafkaClient`. This is used in `socket.create_connection` to set the source of the connection.

Related to #219

### CODE REVIEW

File: pykafka/broker.py [hunk 1]
```diff
@@ -46,7 +46,9 @@ def __init__(self,
                  handler,
                  socket_timeout_ms,
                  offsets_channel_socket_timeout_ms,
-                 buffer_size=1024 * 1024):
+                 buffer_size=1024 * 1024,
+                 source_host='',
+                 source_port=0):
```
Comment (kbourgoin) [inline pykafka/broker.py]: Such weird defaults....  (yes, i know they're the socket defaults. just weird.)

File: pykafka/connection.py [hunk 2]
```diff
@@ -63,7 +76,8 @@ def connect(self, timeout):
         """Connect to the broker."""
         self._socket = socket.create_connection(
             (self.host, self.port),
-            timeout=timeout / 1000
+            timeout / 1000,
+            self.source_host, self.source_port
```
Comment (kbourgoin) [inline pykafka/connection.py]: If these are kwargs, that would be preferable to using over positional.

**Comment by kbourgoin** [comment]
Travis failed, so there's something up with the socket creation. Once that's fixed, this looks good to me. Might want to put a link to the python documentation for that setting in a docstring (maybe the client?) so that it's obvious what it is. It's not a setting I've seen before, so it might help other people too.

File: pykafka/connection.py (hunk 2 above)
Comment (emmett9001) [inline pykafka/connection.py]: [They don't appear to be kwargs](https://docs.python.org/2/library/socket.html#socket.create_connection)

Comment (kbourgoin) [inline pykafka/connection.py]: I was afraid that was the case. The stdlib docs are super confusing about that sometimes and functions will sometimes accept the kwarg when they shouldn't.

**Pull request merged**